EIPS/eip-5380.md

```diff
@@ -0,0 +1,99 @@
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### `IERC5380`
+
+```solidity
+/// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0;
+
+interface IERC5380 is IERC165 {
+    /// @notice Emitted when the amount of entitlement a user changes. If user is the zero address, then the user is the owner
+    event EntitlementChanged(address indexed user, address indexed contract, uint256 indexed tokenId);
+
+    /// @notice             Set the user associated with the given EIP-721 token as long as the owner is msg.sender.
+    /// @dev                SHOULD NOT revert if the owner is not msg.sender.
+    /// @param  user        The user to grant the entitlement to
+    /// @param  contract    The property to grant
+    /// @param  tokenId     The tokenId to grant the properties of
+    function entitle(address user, address contract, uint256 tokenId) external;
+
+    /// @notice             Set the user associated with the given EIP-721 token as long as the owner is msg.sender.
+    /// @dev                SHOULD NOT revert if the owner is not msg.sender.
+    /// @param  user        The user to grant the entitlement to
+    /// @param  contract    The property to grant
+    /// @param  tokenId     The tokenId to grant the properties of
+    /// @param  reason      The reason for the entitlement
+    function entitle(address user, address contract, uint256 tokenId, bytes calldata reason) external;
+
+    /// @notice             Get the maximum number of users that can receive this entitlement
+    /// @param  contract    The contract to query
+    /// @param  tokenId     The tokenId to query
+    function maxEntitlements(address contract, uint256 tokenId) external view (uint256 max);
+
+    /// @notice             Get the user associated with the given contract and tokenId.
+    /// @dev                Defaults to contract.ownerOf(tokenId)
+    /// @param  contract    The contract to query
+    /// @param  tokenId     The tokenId to query
+    function entitlementOf(address contract, uint256 tokenId) external view returns (address user);
+}
+```
+
+`supportsInterface` MUST return true when called with `IERC5380`'s interface ID.
```

Comment:
Generally I recommend including the interface id explicitly in the EIP. Helps to catch bugs when copying the interface.

Reply:
Personally, I prefer keeping it this way while the EIP is in draft. When it moves to review I'd agree that it's probably best to explicitly state it.